src/theia/sfm/feature_extractor_and_matcher.cc

```diff
@@ -63,8 +63,24 @@ void ExtractFeatures(
     const FeatureExtractorAndMatcher::Options& options,
     const std::string& image_filepath,
     std::vector<Keypoint>* keypoints,
-    std::vector<Eigen::VectorXf>* descriptors) {
+    std::vector<Eigen::VectorXf>* descriptors,
+    const std::string& imagemask_filepath = "") {
   std::unique_ptr<FloatImage> image(new FloatImage(image_filepath));
+  std::unique_ptr<FloatImage> image_mask(new FloatImage(imagemask_filepath));
+
+  if (imagemask_filepath != "") {
+  // Check the size of the image and its associated mask.
+    if (image_mask.get()->Width() != image.get()->Width() ||
```

Comment:
Indeed, It's cleaner. However, it implies to verify Wight and Height separately and copy the same message twice:

`CHECK_EQ(image_mask->Width(), image->Width()) << "Massive error message about size." 
 CHECK_EQ(image_mask->Height(), image->Height()) << "Massive error message about size."`

Would you have another trick to avoid it?